client/app/bundles/course/surveys/components/NewSurveyButton.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { PropTypes } from 'react';
+import { injectIntl, defineMessages } from 'react-intl';
+import FloatingActionButton from 'material-ui/FloatingActionButton';
+import ContentAdd from 'material-ui/svg-icons/content/add';
+import SurveyFormContainer from '../containers/SurveyFormContainer';
+
+const translations = defineMessages({
+  newSurvey: {
+    id: 'course.surveys.newSurveyForm.title',
+    defaultMessage: 'New Survey',
+  },
+  success: {
+    id: 'course.surveys.newSurveyForm.success',
+    defaultMessage: 'Survey "{title}" created.',
+  },
+  failure: {
+    id: 'course.surveys.newSurveyForm.failure',
+    defaultMessage: 'Failed to create "{title}".',
+  },
+});
+
+const styles = {
+  floatingButton: {
+    position: 'fixed',
+    bottom: 40,
+    right: 40,
+  },
+};
+
+const propTypes = {
+  createSurvey: PropTypes.func.isRequired,
+  showSurveyForm: PropTypes.func.isRequired,
+  intl: PropTypes.shape({
+    formatMessage: PropTypes.func.isRequired,
+  }).isRequired,
+  createPath: PropTypes.string,
+};
+
+class NewSurveyButton extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.createSurveyHandler = this.createSurveyHandler.bind(this);
+  }
+
+  createSurveyHandler(data, resetForm, invalidateForm) {
+    const { createSurvey, intl, createPath } = this.props;
+    const payload = { survey: data };
+    const successMessage = intl.formatMessage(translations.success, data);
+    const failureMessage = intl.formatMessage(translations.failure, data);
+
+    createSurvey(createPath, payload, invalidateForm, successMessage, failureMessage);
+  }
+
+  render() {
+    const { intl, showSurveyForm, createPath } = this.props;
+
+    if (!createPath) {
+      return <div />;
+    }
+
+    // Default start date to start of next day in current Time Zone
+    const defaultStartAt = new Date();
```

Comment:
Just wondering, is it beneficial to move some of these default form values into a separate file/helper so we can set the defaults in just 1 file?